src/snowflake/cli/api/secure_utils.py

```diff
@@ -27,3 +80,39 @@ def file_permissions_are_strict(file_path: Path) -> bool:
         | stat.S_IXOTH  # executable by others
     )
     return (file_path.stat().st_mode & accessible_by_others) == 0
+
+
+def file_permissions_are_strict(file_path: Path) -> bool:
+    if IS_WINDOWS:
+        return _windows_file_permissions_are_strict(file_path)
+    return _unix_file_permissions_are_strict(file_path)
+
+
+def chmod(path: Path, permissions_mask: int) -> None:
```

Comment:
That would cause import loop, `secure_path` uses this function :)

Reply:
It is also used in `restrict_permissions` (originally here), which is used by SecurePath (for example in `.touch()`)
We would have move all this module into SecurePath, I thought it would be messy